src/components/navigator/views/room-settings/NavigatorRoomSettingsBasicTabView.tsx

```diff
@@ -1,30 +1,52 @@
 import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
-import { RoomDeleteComposer } from '@nitrots/nitro-renderer';
+import { RoomDeleteComposer, RoomSettingsSaveErrorEvent } from '@nitrots/nitro-renderer';
 import { FC, useEffect, useState } from 'react';
-import { CreateLinkEvent, GetMaxVisitorsList, IRoomData, LocalizeText, SendMessageComposer } from '../../../../api';
+import { CreateLinkEvent, GetMaxVisitorsList, IRoomData, LocalizeText, RoomSettingsErrorType, SendMessageComposer } from '../../../../api';
 import { Base, Column, Flex, Text } from '../../../../common';
-import { useNavigator, useNotification } from '../../../../hooks';
+import { useMessageEvent, useNavigator, useNotification } from '../../../../hooks';
 
 const ROOM_NAME_MIN_LENGTH = 3;
 const ROOM_NAME_MAX_LENGTH = 60;
 const DESC_MAX_LENGTH = 255;
+const TAGS_MAX_LENGTH = 15;
 
 interface NavigatorRoomSettingsTabViewProps
 {
     roomData: IRoomData;
-    handleChange: (field: string, value: string | number | boolean) => void;
+    handleChange: (field: string, value: string | number | boolean | string[]) => void;
     onClose: () => void;
 }
 
-
 export const NavigatorRoomSettingsBasicTabView: FC<NavigatorRoomSettingsTabViewProps> = props =>
 {
     const { roomData = null, handleChange = null, onClose = null } = props;
     const [ roomName, setRoomName ] = useState<string>('');
     const [ roomDescription, setRoomDescription ] = useState<string>('');
+    const [ roomTag1, setRoomTag1 ] = useState<string>('');
+    const [ roomTag2, setRoomTag2 ] = useState<string>('');
+    const [ tagIndex, setTagIndex ] = useState(0);
+    const [ typeError, setTypeError ] = useState<string>('');
     const { showConfirm = null } = useNotification();
     const { categories = null } = useNavigator();
 
+    useMessageEvent<RoomSettingsSaveErrorEvent>(RoomSettingsSaveErrorEvent, event =>
+    {
+        const parser = event.getParser();
+
+        if (!parser) return;
+
+        switch (parser.code)
+        {
+            case RoomSettingsErrorType.ROOM_TAGS_BADWWORDS:
+            case RoomSettingsErrorType.RESTRICTED_TAGS:
```

Comment:
use the values provided in `RoomSettingsSaveErrorParser` (for example `RoomSettingsSaveErrorParser.ERROR_INVALID_TAG`. Also make sure you use the correct localization for each error type (you can check the localization for each error type here: https://github.com/dank074/HabboSwfOpenSource/blob/5bad64a6704e38576c96c08a7a3a246305e31a92/src/com/sulake/habbo/navigator/roomsettings/RoomSettingsCtrl.as#L336)